### DISCUSSION TITLE
(gpg4win) Include parameter to install all features

### DISCUSSION THREAD

**Discussion by necrose99** [chocolatey-community/chocolatey-packages#1799] [Ideas]
choco install gpg4win --params **"full"** would be nice as to install **all features**

various  features would be nice to have ,  --params  "feature_Name"  to add or **full/all** for everything 
 github gpg signing is bit easier with gpg agent etc... 

i find myself having to manually re-installing to get the wanted features .....

**Comment by AdmiringWorm** [comment]
That is a great suggestion, unfortunately, the installer doesn't make it easy to allow that to happen.

While the installers have a couple of arguments that could be used (the installer itself have the arguments `WITH_BROWSER` and `MINIMAL`), but nothing related to enabling any of the other components or all of the components.

While we could possibly manipulate the registry before running the installer, we would never be able to guarantee that there would be possible to enable everything, as such we'd need to think a bit on this one before we can accept the idea.

**Reply by necrose99** [reply]
[Auto Hotkey](https://www.autohotkey.com/)  Alt N / next i suppose ,  its what chocolatey has used for quite some time to click drive menus. 
https://www.macrocreator.com/ https://github.com/Pulover/PuloversMacroCreator another that more gui based and can record key/mouse clicks to a macro script/s in ahk scipts. 

https://community.chocolatey.org/packages/autohotkey 
map menu options ie 

- kleopatra --params "kleopatra" 
- gpa
- gpgOL
- gpgEX
- Browser Integration {browser} 
- ALL /full --params "full"  alias all 


![gpg4win](https://user-images.githubusercontent.com/2396239/154106381-1cf8469a-65af-4129-afc0-bd2f87079fa4.png)

**Reply by TheCakeIsNaOH** [reply]
In this case, the installer is run with a silent switch, not automated with autohotkey.
https://github.com/chocolatey-community/chocolatey-packages/blob/master/automatic/gpg4win/tools/chocolateyInstall.ps1#L11

Silent switches are much preferred, as they work on systems without GUIs and when remoting in, and they are much more reliable, especially across software versions.

**Reply by necrose99** [reply]
i've opened with upstream , or soon will to add switches as it would do better .. 

else default of silent , 
(else) if params match case : 
i suppose i could make out some ahk's with the macro recorder as a tentative measure , however i pray upstream will bake them in for the long haul.

**Reply by AdmiringWorm** [reply]
Using AHK for this package would be a no-go.

AutoHotkey is used for packages that use installers that can not be run in an automated way (IE, they are missing silent arguments, or the silent arguments do not work as intended).

This package does not fall under that category, and there would not be any benefits with using autohotkey in this case as the exact same thing could be achieved using a few registry hacks.

**Reply by necrose99** [reply]
https://dev.gnupg.org/T5908
requested switches from upstream as a Wishlist  , as your existing install silent and or via 
/gpa or feature names /all for everything could be passed far eassier... 
and if new features added utils /"newi-tem"-name-flag..... ??? what ever it happens to actually be, 

ie  **Gpg4win)--params "/all"**
